include/nbl/ext/CentralLimitBoxBlur/CBlurPerformer.h

```diff
@@ -0,0 +1,131 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_EXT_BLUR_C_BLUR_PERFORMER_INCLUDED_
+#define _NBL_EXT_BLUR_C_BLUR_PERFORMER_INCLUDED_
+
+#include "nabla.h"
+
+namespace nbl
+{
+namespace ext
+{
+namespace CentralLimitBoxBlur
+{
+
+typedef uint32_t uint;
+struct alignas(16) uvec4
+{
+    uint x, y, z, w;
+};
+#include "nbl/builtin/glsl/ext/CentralLimitBoxBlur/parameters_struct.glsl"
+
+class CBlurPerformer final : public core::IReferenceCounted
+{
+public:
+    _NBL_STATIC_INLINE_CONSTEXPR uint32_t DEFAULT_WORKGROUP_SIZE = 256u;
+    _NBL_STATIC_INLINE_CONSTEXPR uint32_t PASSES_PER_AXIS = 3u;
+
+    typedef nbl_glsl_ext_Blur_Parameters_t Parameters_t;
+
+    struct DispatchInfo_t
+    {
+        uint32_t wg_count[3];
+    };
+
+    CBlurPerformer(video::IVideoDriver* driver, uint32_t maxDimensionSize, bool useHalfStorage);
+
+    static core::SRange<const asset::SPushConstantRange> getDefaultPushConstantRanges();
+
+    static inline void defaultBarrier()
+    {
+        video::COpenGLExtensionHandler::extGlMemoryBarrier(GL_SHADER_STORAGE_BARRIER_BIT);
+    }
```

Comment:
you need to provide barrier APIs similar to the blit extension

Reply:
I don't think the blit API has this.
[Do we want to take a similar route to `CScanner`?](https://github.com/Devsh-Graphics-Programming/Nabla/blob/9b5b2a443608d2fc1fe4da8af85947961cae4e0a/include/nbl/video/utilities/CScanner.h#L371)

Reply:
yes please, but what does the blit API have for helping with barriers then?